numba/np/arrayobj.py

```diff
@@ -5278,6 +5280,13 @@ def codegen(context, builder, sig, args):
         itemsize = get_itemsize(context, aryty)
         ll_itemsize = Constant(buf.itemsize.type, itemsize)
         nbytes = builder.mul(buf.nitems, buf.itemsize)
+        ll_offset_size = builder.mul(arg_offset, ll_itemsize)
+        nbytes = builder.sub(nbytes, ll_offset_size)
```

Comment:
`nbytes` can be negative as well, right? If the offset * itemsize > nbytes, sub will be a negative number.

NumPy raises an exception in those cases:
```
import numpy as np
from numba import njit

@njit
def foo():
    s = b'hello world'
    return np.frombuffer(s, dtype='S1', count=5, offset=16)

print(foo())  # print [b'1' b'd' b'_' b'n' b'j']
print(foo.py_func())  # raise ValueError
```